### PR TITLE
Save friend trees with Electron ID SF

### PR DESCRIPTION
**Pull request by previsualconsent**
2 assumptions made:
if energy branch or eta == -999, then the SF = 0
if pT < 10 GeV use SF for 10 GeV

### CODE REVIEW

File: ZFitter/src/addBranch_class.cc [hunk 1]
```diff
@@ -21,18 +22,84 @@ addBranch_class::~addBranch_class(void)
...
+
+	std::cout << gDirectory->GetName() << std::endl;
+	TTree* newtree = new TTree(treename, treename);
+	newtree->SetDirectory(gDirectory);
+	Float_t EleIDSF[] = {1,1,1};
+	newtree->Branch(branchname, EleIDSF, branchname + "[3]/F");
+
+	TH2F * sf = NULL;
+	if(branchname.Contains("veto")) {
+		sf = (TH2F *) TFile::Open("/eos/project/c/cms-ecal-calibration/data/EleIDSF/Moriond_cut-based_80XID_veto.root")->Get("EGamma_SF2D");
```
Comment (shervin86) [inline ZFitter/src/addBranch_class.cc]: can we make the SF branch name to be: ```eleIDSF_<ID>``` ?

File: ZFitter/src/addBranch_class.cc [hunk 2]
```diff
@@ -21,18 +22,84 @@ addBranch_class::~addBranch_class(void)
...
 	if(BranchName.Contains("ZPt"))   return AddBranch_ZPt(originalChain, treename, BranchName.ReplaceAll("ZPt_", ""), fastLoop);
 	if(BranchName.CompareTo("LTweight") == 0) return AddBranch_LTweight(originalChain, treename);
+	if(BranchName.Contains("EleIDSF")) return AddBranch_EleIDSF(originalChain, treename, BranchName, energyBranchName, isMC);
 	std::cerr << "[ERROR] Request to add branch " << BranchName << " but not defined" << std::endl;
 	return NULL;
 }
 
+TTree* addBranch_class::AddBranch_EleIDSF(TChain* originalChain, TString treename, TString branchname, TString energyBranchName, bool isMC)
+{
+	Float_t etaSCEle[] = {0,0,0};
```
Comment (shervin86) [inline ZFitter/src/addBranch_class.cc]: if is MC, return an empty tree

Comment (previsualconsent) [inline ZFitter/src/addBranch_class.cc]: if is not MC?

File: ZFitter/src/addBranch_class.cc (hunk 1 above)
Comment (previsualconsent) [inline ZFitter/src/addBranch_class.cc]: yes.

File: ZFitter/src/addBranch_class.cc (hunk 2 above)
Comment (shervin86) [inline ZFitter/src/addBranch_class.cc]: if not go on with what is already there, I was just suggesting to move the if(MC) before and skip many things

**Comment by shervin86** [comment]
if possible, squash the commits

File: ZFitter/src/addBranch_class.cc (hunk 2 above)
Comment (previsualconsent) [inline ZFitter/src/addBranch_class.cc]: Actually, that makes ZFitter error, if the lengths of the friends are the wrong size. None of the other addBranch functions do that.

Comment (shervin86) [inline ZFitter/src/addBranch_class.cc]: ok

**Pull request merged**